src/solvers/file_solution_history.C

```diff
@@ -77,9 +88,16 @@ void FileSolutionHistory::find_stored_entry(Real time)
   {
     stored_sols = lower_bound_it_decremented;
   }
-  else
+  else // Neither of the two candidate keys matched our time
   {
-    libmesh_error_msg("Failed to set stored solutions iterator to a valid value.");
+    if(storing) // If we are storing, this is fine, we need to create a new entry, so just return
+    {
+      return;
```

Comment:
Same question here - are we just getting away with this because we only create new entries at the end of the history?

Reply:
The only time find_stored_entry needs to change stored_sols is if we actually manage to find an existing entry for that time. This should never happen during a timestepping loop unless erase has not been to remove garbage entries. 

This could happen if we embed the timestepping loop inside an adaptive or continuation loop for example. So perhaps we should add a method to clear the stored_solutions map ? This would also delete old files ideally.